### PR TITLE
Nodejs-week2

### PR DESCRIPTION
**Pull request by Yuliia-Balandiuk**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -1,14 +1,79 @@
...
-const knex = require("../database");
+const knex = require('../database');
 
-router.get("/", async (request, response) => {
+router.get('/', async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const allMeals = await knex.select().table('meal');
```
Comment (Divyajg) [inline src/backend/api/meals.js]: You should include something in `select(-xx-)`

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -1,14 +1,79 @@
...
+
+    if (allMeals.length !== 0) {
+      res.send(allMeals);
+    } else {
+      res.status(404).json({ error: 'Meals not found' });
```
Comment (Divyajg) [inline src/backend/api/meals.js]: There is a possibility for `allMeals` to be none, so it won't be an error and hence the status won't be 404. Instead, the status can be 200 and send a message saying no meals found.

**Review by Divyajg — COMMENTED** [review]
Looks perfect, but I have only two comments, that are applicable in several places in all meals, reservations and app.js files.
Keep coding.

File: src/backend/api/meals.js [hunk 3]
```diff
@@ -1,14 +1,79 @@
...
+    }
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.put('/:id', async (req, res) => {
+  try {
+    const mealBody = req.body;
+    const mealId = req.params.id;
```
Comment (viedamayanti) [inline src/backend/api/meals.js:47]: Nice one, I get an idea from your code to write dynamic meal id as I hardcode my code 😅

File: src/backend/api/meals.js [hunk 4]
```diff
@@ -1,14 +1,79 @@
...
+    const meal = await knex('meal').where({ id: mealId });
+
+    if (meal.length === 0) {
```
Comment (viedamayanti) [inline src/backend/api/meals.js:50]: You also put many if statement, I think more detail than mine. Well done Julia 💪

**Pull request merged**